secure-mobile-push/smp-android-client/securemobilepush/src/main/java/de/adorsys/android/securemobilepush/sms/SmsTool.java

```diff
@@ -0,0 +1,33 @@
+package de.adorsys.android.securemobilepush.sms;
+
+import android.Manifest;
+import android.app.Activity;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.v4.content.ContextCompat;
+
+public class SmsTool {
+    public static final int REQUEST_CODE_ASK_PERMISSIONS = 123;
+
+    @NonNull private Context context;
+
+    public SmsTool(@NonNull Context context) {
+        this.context = context;
+    }
+
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    public void requestSMSPermission() {
+        Activity activity = (Activity) context;
+        final String permission = Manifest.permission.RECEIVE_SMS;
+        int hasSpecificPermission = ContextCompat.checkSelfPermission(context, permission);
+        if (hasSpecificPermission != PackageManager.PERMISSION_GRANTED) {
+            if (!activity.shouldShowRequestPermissionRationale(permission)) {
```

Comment:
Fixed in commit: https://github.com/adorsys/secure-key-storage/commit/fd9ee5c6e8926d9e0ff95c5c30b8531ff5143d8b